lib/MockXMLHttpRequest.js

```diff
@@ -158,40 +159,45 @@ MockXMLHttpRequest.prototype.send = function(data) {
 
     var response = MockXMLHttpRequest.handle(new MockRequest(self));
 
-    if (response && response instanceof MockResponse) {
+    Promise.resolve(response).then(function(response) {
```

Comment:
Hm, I think if request wasn't handled by promise it should be rejected. In other case this promise will be resolved, when passed promise resolved as well.

I tried to do postponed call in `xhr-mock-proxy-handler` using setTimeout with 5 seconds awaiting. And run written tests, tests passed.

If I incorrectly understood you, please correct me.